android/src/main/java/com/bradbumbalough/RCTTwilioChat/RCTTwilioChatClient.java

```diff
@@ -294,7 +294,16 @@ public void onConnectionStateChange(ChatClient.ConnectionState state) {
 
     @Override
     public void onChannelAdd(final Channel channel) {
-        sendEvent("chatClient:channelAdded", RCTConvert.Channel(channel));
+        if (RCTTwilioChatClient.getInstance().client.getProperties().getSynchronizationStrategy() == ChatClient.SynchronizationStrategy.ALL) {
```

Comment:
The drawback to always calling synchronize is that it may perform extra calls for people using SynchronizationStrategyChannelList that they may not want. Not sure how important it is to account for that here, but that's why I added the if statement. I'm ok with just calling every time if you guys think that's ok.